include/ggml.h

```diff
@@ -1801,6 +1802,17 @@ extern "C" {
             int                   p0,
             int                   p1);
 
+    // Move tensor elements by an offset given for each dimension. Elements that
+    // are shifted beyond the last position are wrapped around to the beginning.
+    GGML_API struct ggml_tensor * ggml_roll(
+            struct ggml_context * ctx,
+            struct ggml_tensor  * a,
+            int                   shift0,
+            int                   shift1,
+            int                   shift2,
+            int                   shift3);
+
```

Comment:
Should we change the signature like so:

```suggestion
    // Move tensor elements by an offset given for each dimension. Elements that
    // are shifted beyond the last position are wrapped around to the beginning.
    // If tensor b is provided, the shifts become: shiftX' = shiftX + b[X]
    GGML_API struct ggml_tensor * ggml_roll(
            struct ggml_context * ctx,
            struct ggml_tensor  * a,
            struct ggml_tensor  * b, // optional I64 [GGML_MAX_DIMS]
            int                   shift0,
            int                   shift1,
            int                   shift2,
            int                   shift3);

```

This in theory would allow to pass the shifts as dynamic graph input data, which makes the graph static in terms of nodes (and op params). Could be useful for graph reuse.

Reply:
I wouldn't bother with this unless there is some use case that you are thinking about, otherwise this is likely to end increasing complexity for no real reason. If you decide to add this, for clarity I would recommend adding a different function that takes only the tensor, rather than trying to do everything with a single function and optional parameters.